subprojects/ide-native/src/main/groovy/org/gradle/ide/visualstudio/tasks/GenerateFiltersFileTask.java

```diff
@@ -44,9 +45,16 @@ public void setVisualStudioProject(VisualStudioProject vsProject) {
         this.visualStudioProject = (DefaultVisualStudioProject) vsProject;
     }
 
-    @Nested
-    public VisualStudioProject getVisualStudioProject() {
-        return visualStudioProject;
+    /**
+     * Returns the xml string that will be written to the filter file.
+     *
+     * @since 4.7
+     */
+    @Input
+    public String getFilterFileXml() {
```

Comment:
I think this isn't quite what Adam had in mind.  It seems really weird that the input to this task is the output of this task.

I think instead we would change how we generate the file:
https://github.com/gradle/gradle/blob/189a80a645ee42c1bb3443894a6b74fe1a6c6f15/subprojects/ide/src/main/java/org/gradle/plugins/ide/api/GeneratorTask.java#L99

Either we would write into a `ByeArrayOutputStream` and compare that to the contents of the output file and rewrite the file (if different) or we would write the contents to a temporary file, compare it and then move it over (if different).

We can set `setDidWork(false)` in the case that the file does not need to be changed.

That gets rid of the extra input property on each of the tasks.

Reply:
I was basing this off the comment [here](https://github.com/gradle/gradle/pull/4531#discussion_r173016544).  Just doing this as part of the task action itself is a further simplification and how we ultimately want to handle it - this is tracked [here](https://github.com/gradle/gradle-native/issues/576).  I'm not sure whether it's still worthwhile to merge this version of things or not.

Reply:
Let's leave this open/close it and just handle the rest of it in https://github.com/gradle/gradle-native/issues/576